experimental/cache_analysis.py

```diff
@@ -0,0 +1,139 @@
+from matplotlib.pyplot import figure
+import matplotlib.pyplot as plt
+import io
+import numpy as np
+import pandas as pd
+from statistics import mean, median
+import collections
+import csv
+import seaborn as sns
+
+""" This functions draws two types of plots for cases where each x value has only one y value.
```

Comment:
nit, but I'd try to phrase this in the style of the Python style guide: 

```py
"""Draws two types of plots [...]

Args:
  data: <description of what data is -- e.g. its dimensions>
  name: The name of the figure.
"""
```

https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings